fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/OnlineIndexer.java

```diff
@@ -1015,11 +1022,19 @@ public void rebuildIndex(@Nonnull FDBRecordStore store) {
     @Nonnull
     private CompletableFuture<TupleRange> buildEndpoints(@Nonnull FDBRecordStore store, @Nonnull RangeSet rangeSet,
                                                          @Nullable AtomicLong recordsScanned) {
+        boolean isIdempotent = store.getIndexMaintainer(index).isIdempotent();
+        final IsolationLevel isolationLevel =
+                isIdempotent ?
+                // if idempotent: since double indexing is harmless, we can skip the range conflict protection. At worse,
+                // some records will be re-indexed.
+                IsolationLevel.SNAPSHOT :
+                IsolationLevel.SERIALIZABLE;
         final ExecuteProperties limit1 = ExecuteProperties.newBuilder()
                 .setReturnedRowLimit(1)
-                .setIsolationLevel(IsolationLevel.SERIALIZABLE)
+                .setIsolationLevel(isolationLevel)
```

Comment:
Even though only one record is scanned, I think that record should be added to read conflict.

I wouldn't worry about reproducing it from tests though.